lib/next_ls/helpers/ast_helpers.ex

```diff
@@ -148,4 +148,228 @@ defmodule NextLS.ASTHelpers do
+    defp prewalk({:->, meta, [left, _right]} = ast, acc) do
+      acc = increase_scope_nesting(acc, meta[:line])
+      acc = find_symbols(left, acc)
+      {ast, acc}
+    end
+
+    # special case for 'cond'
+    defp prewalk({:cond, meta, _args} = ast, acc) do
+      acc = increase_scope_nesting(acc, "cond#{meta[:line]}")
+      {ast, acc}
+    end
+
+    # increase scope on enter
+    defp prewalk({operation, meta, _args} = ast, acc) when operation in @scope_breaks do
+      acc = increase_scope_nesting(acc, meta[:line])
+      {ast, acc}
+    end
+
+    # special case for 'cond'
+    defp prewalk({:do, _args} = ast, %{scope: ["cond" <> _ | _]} = acc) do
+      acc = increase_scope_nesting(acc, "conddo")
+      {ast, acc}
+    end
+
+    # increase scope on enter 'do/end' block
+    defp prewalk({operation, _args} = ast, acc) when operation in @blocks do
+      acc = increase_scope_nesting(acc, operation)
+      {ast, acc}
+    end
+
+    # search symbols inside left side of a match or <- and fix processig sequence
+    defp prewalk({operation, meta, [left, right]}, acc) when operation in [:=, :<-, :destructure] do
+      acc = find_symbols(left, acc)
+      {{operation, meta, [right, left]}, acc}
+    end
+
+    # exclude attribute macro from variable search
+    defp prewalk({:@, _, _}, acc) do
+      {nil, acc}
+    end
+
+    # find variable
+    defp prewalk({name, meta, nil} = ast, acc) do
+      range = calculate_range(name, meta[:line], meta[:column])
+      type = if range in acc.sym_ranges, do: :sym, else: :ref
+      var = {type, name, range, acc.scope}
+
+      acc = collect_var(acc, var)
+
+      {ast, acc}
+    end
+
+    defp prewalk(ast, acc), do: {ast, acc}
+
+    # decrease scope when exiting it
+    defp postwalk({operation, _, _} = ast, acc) when operation in @scope_ends do
+      acc = decrease_scope_nesting(acc)
+      {ast, acc}
+    end
+
+    # decrease scope when exiting 'do/else' block
+    defp postwalk({operation, _} = ast, acc) when operation in @blocks do
+      acc = decrease_scope_nesting(acc)
+      {ast, acc}
+    end
+
+    defp postwalk(ast, acc), do: {ast, acc}
+
+    defp find_symbols(ast, acc) do
+      {_ast, acc} = Macro.prewalk(ast, acc, &find_symbol/2)
+      acc
+    end
+
+    defp find_symbol({operation, _, _}, acc) when operation in [:^, :unquote] do
+      {nil, acc}
+    end
+
+    # exclude right side of 'when' from symbol search
+    defp find_symbol({:when, _, [left, _right]}, acc) do
+      {left, acc}
+    end
+
+    defp find_symbol({name, meta, nil} = ast, acc) do
+      range = calculate_range(name, meta[:line], meta[:column])
+      acc = Map.update!(acc, :sym_ranges, &[range | &1])
+      {ast, acc}
+    end
+
+    defp find_symbol(ast, acc), do: {ast, acc}
+
+    defp calculate_range(name, line, column) do
+      length = name |> to_string() |> String.length()
+
+      {line..line, column..(column + length)}
```

Comment:
I think the column range might need a `- 1`

Reply:
yes, you are right, will fix